assets/src/components/selection/datetime-select/index.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * External dependencies
+ */
+import { stringify } from 'querystringify';
+import moment from 'moment';
+import { isUndefined, pickBy, isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import { __ } from '@eventespresso/i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { Placeholder, SelectControl, Spinner } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { Component, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { datetimeSelectOptions } from './options';
+
+const nowDateAndTime = moment();
+
+/**
+ * DatetimeSelect component.
+ * A react component for an datetime selector.
+ *
+ * @param {Array} datetimes  An empty array or array of Datetime Entities. See
+ *                          prop-types for shape.
+ * @param {function} onDatetimeSelect  The callback on selection of datetime.
+ * @param {string} selectLabel The label for the select input.
+ * @param {number} selectedDatetimeId  If provided, the id of the datetime to
+ *   pre-select.
+ * @param {number} forEventId   ID for Event to retrieve datetimes from
+ * @param {boolean} isLoading  Whether or not the selector should start in a
+ *                               loading state
+ * @return {Function}  A pure component function.
+ * @constructor
+ */
+class DatetimeSelect extends Component
+{
+	constructor() {
+		super( ...arguments );
+		this.placeHolder = this.placeHolder.bind( this );
+	}
+
+	placeHolder() {
+		const { isLoading, selectLabel } = this.props;
+		return (
+			<Fragment>
+				<Placeholder
+					icon="calendar"
+					label={ selectLabel }
+				>
+					{ ! isLoading ?
+						__(
+							'There are no datetimes to select from. You need' +
+							' to create a datetime first.',
```

Comment:
I know I've done this as well in the js and I _think_ its okay, however we'll have to make sure that the Webpack extraction plugin grabs these strings okay (I keep forgetting to double-check).

Reply:
heh... i think my IDE did that on its own...

Reply:
I'm not seeing any of the translations from this component getting picked up. Even the ones that are not concatenated.

Is there something I have to do other than `npm run build` to get them into `translation-map.json` ?

Reply:
hmm... just running `npm run build` _should_ be sufficient.   If they aren't getting picked up though then it's possible there's a yet undiscovered bug with the extractor I built.